src/biogeochem/CNVegComputeSeedMod.F90

```diff
@@ -118,6 +121,7 @@ subroutine ComputeSeedAmounts(bounds, &
           my_deadstem_seed = 0._r8
 
           pft_type = patch%itype(p)
+          leafcn = leafcn_t_evolving(p)
```

Comment:
from meeting with @ekluzek 
the local variable is degrading efficiency, though we checked and found that this happens once per year in transient simulations, so it's not a huge deal in this case.

Reply:
If I decided to revert this change, then it may make sense to remove all (or most, e.g. not pft_type) the local variables.

Reply:
...but that's a refactor outside of this PR's scope

Reply:
so i'm resolving